### PR TITLE
Update README for 2.4.1 release

### PR DESCRIPTION
**Pull request by mattdangerw**
Won't actually merge this until the pip release is out!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,32 +64,22 @@ Fine-tune BERT on a small sentiment analysis task using the
 
 ```python
 import keras_nlp
-from tensorflow import keras
 import tensorflow_datasets as tfds
 
-imdb_train, imdb_test = tfds.load(
+dataset = tfds.load(
```
Comment (jbischof) [inline README.md:69]: Can we keep the test set? I think it's bad practice to fit any model without it.

File: README.md [hunk 2]
```diff
@@ -47,14 +47,14 @@ We are a new and growing project and welcome [contributions](CONTRIBUTING.md).
 To install the latest official release:
 
 ```
-pip install keras-nlp tensorflow --upgrade
```
Comment (jbischof) [inline README.md:50]: Why don't we need to upgrade anymore?

File: README.md (hunk 1 above)
Comment (mattdangerw) [reply README.md:69]: Sure! I like the short punchy version, but not a big deal.

File: README.md (hunk 2 above)
Comment (mattdangerw) [reply README.md:50]: Hmm, yeah let me leave the `--upgrade`. It is no longer needed to update TensorFlow as of https://github.com/keras-team/keras-nlp/pull/697, but useful to make sure you bump your keras-nlp install to the latest.

**Review by jbischof — APPROVED** [review]
Thank you!

**Pull request merged**